runtime/nvqir/qpp/QppCircuitSimulator.cpp

```diff
@@ -135,6 +135,28 @@ class QppCircuitSimulator : public nvqir::CircuitSimulatorBase<double> {
     return;
   }
 
+  /// @brief Override for qubit allocation given a state vector.
+  void
+  addQubitsToState(std::size_t count,
+                   std::vector<std::complex<double>> &inputState) override {
+    if (count == 0)
+      return;
+
+    if (state.size() == 0) {
+      // If this is the first time, allocate the entire system
+      // state as the user provided state
+      state = qpp::ket::Map(inputState.data(), stateDimension);
```

Comment:
We take the user input state and map it to a ket. Then, depending on if a state vector was already allocated, the user ket is either:
1. Set as the `state` variable
2. Gets a Kronecker product onto `state`